### PR TITLE
Add substreams block ingestor

### PR DESCRIPTION
**Pull request by mangas**
- Substreams ingestor not currently allowed for supported chains
- Chains using substreams protocol can only have substreams endpoints
- Add substream-head-tracker to the project
- Embed substreams-head-tracker in substreams block ingestor

### CODE REVIEW

File: substreams-head-tracker/rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
```
Comment (maoueh) [inline substreams-head-tracker/rust-toolchain.toml:1]: Would we have a way to use the top-level one directly. Only thing I'm wondering is the `wasm32-unknown-unknown` target.

File: node/src/config.rs [hunk 2]
```diff
@@ -530,6 +530,31 @@ impl Chain {
...
+            if has_only_substreams_providers {
+                bail!(
+                    "{} protocol requires an rpc or firehose endpoint defined",
+                    self.protocol
+                );
+            }
+        }
+
+        // When using substreams protocol, only substreams endpoints are allowed
+        if matches!(self.protocol, BlockchainKind::Substreams) {
```
Comment (neysofu) [inline node/src/config.rs:548]: This is equivalent to an `else`, correct?

**Review by neysofu — APPROVED** [review]
I don't think I have the necessary context to thoroughly review this, so it's great to have other eyes on this. This is pretty in line with what I would expect though, LGTM.

File: node/src/config.rs (hunk 2 above)
Comment (mangas) [reply node/src/config.rs:548]: Yes, this was mostly to make it clearer, if you think it isn't I can refactor to improve

Comment (neysofu) [reply node/src/config.rs:548]: No need, it's fine!

**Pull request merged**